metrics/prometheus_exporter.go

```diff
@@ -31,19 +32,28 @@ var (
 	curPromSrvMux sync.Mutex
 )
 
-func newPrometheusExporter(config *metricsConfig, logger *zap.SugaredLogger) (view.Exporter, error) {
+type emptyPromExporter struct{}
+
```

Comment:
Done, though probably unnecessary because the package will fail to compile anyway without it when we try to return an `emptyPromExporter` from one of the `getFactory` functions.